### ISSUE TITLE
The change seems breaking the compatibility with multi-config-files in default situation

### ISSUE THREAD

**Issue by KobeArthurScofield** [v2fly/fhs-install-v2ray#88]
Hello developers,

I used the install-script to install V2Ray on a Debian 10 machine since July and used the default ```v2ray.service``` to start it as a service, and everything worked fine. *And to make the install-script and service unit files up to date before every update, I use a script to fetch the latest install-script from the repository to my machine and then start the install-script automatically, worked fine.*

Then when I tried to update the V2Ray to 4.27.5, something went wrong.  
I started to receive error messages from the machine's COM console when I tried to start it (and really failed to start):
```
Sep  5 19:41:02 ogonj-0 v2ray[19823]: V2Ray 4.27.5 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.14.7 linux/amd64)
Sep  5 19:41:02 ogonj-0 maintainence.sh[19102]: info: Start the V2Ray service.
Sep  5 19:41:02 ogonj-0 v2ray[19823]: A unified platform for anti-censorship.
Sep  5 19:41:02 ogonj-0 v2ray[19823]: main: failed to read config files: [/usr/local/etc/v2ray/config.json] > v2ray.com/core/main/json: failed to execute v2ctl to convert config file. > v2ray.com/core/common/platform/ctlcmd: failed to execute v2ctl:
Sep  5 19:41:02 ogonj-0 v2ray[19823]: v2ctl> Read config:  /usr/local/etc/v2ray/config.json
Sep  5 19:41:02 ogonj-0 v2ray[19823]: panic: open /usr/local/etc/v2ray/config.json: no such file or directory
Sep  5 19:41:02 ogonj-0 v2ray[19823]: goroutine 1 [running]:
Sep  5 19:41:02 ogonj-0 v2ray[19823]: v2ray.com/core/common.Must(...)
Sep  5 19:41:02 ogonj-0 v2ray[19823]: #011v2ray.com/core/common/common.go:26
Sep  5 19:41:02 ogonj-0 v2ray[19823]: v2ray.com/core/infra/control.(*ConfigCommand).Execute(0x104f9b0, 0xc0000301a0, 0x1, 0x1, 0xc00019c2e8, 0xcf59bf1808aea901)
Sep  5 19:41:02 ogonj-0 v2ray[19823]: #011v2ray.com/core/infra/control/config.go:42 +0x673
Sep  5 19:41:02 ogonj-0 v2ray[19823]: main.main()
Sep  5 19:41:02 ogonj-0 v2ray[19823]: #011v2ray.com/core/infra/control/main/main.go:35 +0x217 > exit status 2
```
Of course it could not find the config file because I use multiple config files so no ```config.json``` there.

But it worked fine before:
```
Aug 29 03:54:50 ogonj-0 v2ray[12408]: V2Ray 4.27.4 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.14.7 linux/amd64)
Aug 29 03:54:50 ogonj-0 v2ray[12408]: A unified platform for anti-censorship.
Aug 29 03:54:50 ogonj-0 v2ray[12408]: 2020/08/29 03:54:50 Using confdir from arg: /usr/local/etc/v2ray/
Aug 29 03:54:51 ogonj-0 v2ray[12408]: 2020/08/29 03:54:51 [Info] v2ray.com/core/common/platform/ctlcmd: <v2ctl message>
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/00_log.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/01_api.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/02_dns.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/03_routing.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/04_policy.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/05_inbounds.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/06_outbounds.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/07_transport.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/08_stats.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: v2ctl> Read config:  /usr/local/etc/v2ray/09_reverse.json
Aug 29 03:54:51 ogonj-0 v2ray[12408]: 2020/08/29 03:54:51 [Warning] v2ray.com/core: V2Ray 4.27.4 started
```

When I tried to reinstall the older version it could not work anymore too.

Then I found the service unit file has changed like this:
```
[Unit]
Description=V2Ray Service
After=network.target nss-lookup.target
[Service]
User=nobody
CapabilityBoundingSet=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
AmbientCapabilities=CAP_NET_ADMIN CAP_NET_BIND_SERVICE
NoNewPrivileges=true
Environment=V2RAY_LOCATION_ASSET=/usr/local/share/v2ray/
ExecStart=/usr/local/bin/v2ray -config /usr/local/etc/v2ray/config.json
Restart=on-failure
[Install]
WantedBy=multi-user.target
```
The config path should be a config directory before. It seems a change broke the compatibility.

Is this change is *an accidental change* or *an intended change for some reasons*?

**Comment by IceCodeNew** [comment]
看过您的 profile 判断我可以用中文和你沟通，为了方便我就用中文沟通了。如果您觉得不适我之后会换英文和您交流（不会粤语，抱歉）。

> Is this change is an accident or intended change for some purpose?

这是 #60 引入的变化，这一变化是为了恢复到旧的 v2ray 安装脚本的行为，所以你可以认为这是「intended change for some purpose」。
从维护者的视角来说，本项目之前将 service unit file 配置为适合多配置文件的启动方式，但是没有详尽地说明，造成很多用户在迁移脚本后遇到问题（由于单配置文件和脚本产生的多配置文件共处一个目录）。所以在我看来，#60 是用一个 Breaking change 来修复之前自己引入的 breaking change——但是在我们看来使用多配置文件启动方式的用户更少，且往往更有解决问题的能力，所以才做了 #60 这样的取舍。

首先要在这里为给您造成的不便致歉。

最后关于如果解决这一问题的建议：您可以写一个 drop-in 配置如下：
```bash
mkdir -p '/etc/systemd/system/v2ray.service.d'
echo 'ExecStart=
ExecStart=/usr/local/bin/v2ray -confdir /usr/local/etc/v2ray/conf.d/' > v2ray_execstart_multiconf.conf
```

**Comment by KobeArthurScofield** [comment]
感谢开发者的详细解答，用 drop-in override 解决了目前的问题。

既然是出于各种情况下（实际环境的确什么情况都可能有）的兼容性考虑的话，个人给出的建议是可以在安装脚本或 service unit 文件里面给出提示，或者附加一个适用于多配置文件的 service unit ，当然具体采取哪种方法由你们决定。

> 看过您的 profile 判断我可以用中文和你沟通，为了方便我就用中文沟通了。如果您觉得不适我之后会换英文和您交流（不会粤语，抱歉）。

没有问题的，大家都能看懂就可以，有的时候会有输入法突然故障的尴尬被迫打英文 XD

**Comment by IceCodeNew** [comment]
Closed by #114.
